packages/driver/src/cy/net-stubbing/events/after-response.ts

```diff
@@ -34,14 +36,16 @@ export const onRequestComplete: HandlerFn<NetEventFrames.HttpRequestComplete> =
     if (isAwaitingResponse) {
       // the user is implicitly expecting there to be a successful response from the server, so fail the test
       // since a network error has occured
-      return failCurrentTest(err)
+      throw err
```

Comment:
Might need to check how this interacts with the latest error handling and ensure the right error is displayed. 

I'm of the opinion now that, whenever possible, we should avoid re-throwing errors when we have already caught them and instead wrap them with the context where they're caught and report them via other means (like how `failCurrentTest` calls into `cy.fail`). There's a lot of older code that doesn't follow this pattern, but for new or changed code, it would be nice to have it do so.

Reply:
I think it would be worthwhile to add a test for this error here: https://github.com/cypress-io/cypress/blob/develop/packages/runner/cypress/integration/reporter.errors.spec.js

Reply:
After our pairing session, I added the reporter tests here: [`41e0e79` (#15255)](https://github.com/cypress-io/cypress/pull/15255/commits/41e0e7937a83eb16e982aa57e81898e388659f2c)

There's still the issue with double-wrapping cy.intercept errors, in addition to `cy.fail` propagating to `top` in the reporter tests.